config/v1/types_cluster_version.go

```diff
@@ -707,3 +726,89 @@ type ClusterVersionList struct {
 
 	Items []ClusterVersion `json:"items"`
 }
+
+// UpdateBundle describes updates performed using an optional update bundle.
+type UpdateBundle struct {
+	// enabled indicate if use of an update bundle is enabled. By default it is false, and the
+	// either spec.desiredUpdate.version or spec.desiredUpdate.image fields will be used to
+	// determine the required set of images images, pin and pre-load them before proceeding
+	// to apply the update. Note that in that case no bundle will be used.
+	//
+	// If explicitly set to true then the desiredUpdate.version and desiredUpdate.image fields
+	// will be ignored. Instead the cluster will loop for the update bundle and will use to
+	// determine the version, pin and pre-loaded the images and appy the update.
+	//
+	// +optional
+	Enabled bool `json:"enabled"`
+
+	// monitor indicates if the cluster should monitor device plug events to automatically
+	// detect when an update bundle is available. The default is false.
+	//
+	// +optional
+	Monitor bool `json:"monitor"`
+
+	// file indicates the location of the update bundle when the monitor fields is false. It can
+	// be an absolute file path name or an abosolute block device name.
+	//
+	// +optional
+	File string `json:"file"`
+
+	// node contains the name of the node where the update bundle is available. This is
+	// optional, and by default the cluster will look for the bundle in all the nodes.
```

Comment:
What happens if multiple bundles are found?
Is there any way that this resolution determines that bundles are incompatible? Eg we block upgrades on certain platforms for certain versions, how is this handled with this API?